chains/evm/src/ZKLInbox.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract ZKLInbox {
+    struct Message {
+        string encryptedArweaveLink;
+        string recipientPublicKey;
+        string senderPublicKey;
+        uint256 timestamp;
+    }
+
+    mapping(address => Message[]) private inboxes;
+
+    event MessageSent(address indexed recipient, address indexed sender, uint256 timestamp);
+
+    function sendMessage(
```

Comment:
**[Security]**
- from: @virjilakrum 
```
// TODO: Consider adding a maximum message size limit
function sendMessage(
    address recipient,
    string memory encryptedArweaveLink,
    string memory recipientPublicKey,
    string memory senderPublicKey
) public {
    // ...
}
```
There is no upper limit set for message size. This could lead to potentially very large messages being sent and stored, which could increase gas costs and leave the contract vulnerable to DoS (Denial of Service) attacks. @arhansuba